### PR TITLE
Add last releng images builder prow jobs 

### PR DESCRIPTION
**Pull request by cpanato**
- Add last image build jobs
- turn all presubmits jobs required

k/release associate pr: https://github.com/kubernetes/release/pull/1950

/assign @justaugustus @saschagrunert @hasheddan 

tracking issue: https://github.com/kubernetes/release/issues/1850

### CODE REVIEW

File: config/jobs/image-pushing/releng/k8s-staging-releng.yaml [hunk 1]
```diff
@@ -77,6 +77,7 @@ postsubmits:
         testgrid-dashboards: sig-release-releng-informing, sig-release-image-pushes
         testgrid-alert-email: release-managers+alerts@kubernetes.io
       decorate: true
+      run_if_changed: '^images\/releng\/k8s-ci-builder\/'
```
Comment (cpanato) [inline config/jobs/image-pushing/releng/k8s-staging-releng.yaml:80]: Add this to run only if the image directory change, or there is a reason we want to build every time?

Comment (justaugustus) [reply config/jobs/image-pushing/releng/k8s-staging-releng.yaml:80]:
```suggestion
      run_if_changed: '^(cmd\/|compile-release-tools|go.(mod|sum)|images\/releng\/k8s-ci-builder\/|pkg\/)'
```

Yes, k8s-ci-builder builds all tools in k/release using `./compile-release-tools`: https://github.com/kubernetes/release/blob/fce32c9696f0be229e9898e51cbbe8e497e72027/images/releng/k8s-ci-builder/Dockerfile#L21-L23

...which means any change to the repo packages should be trigger a rebuild.

ref: https://regex101.com/r/cZIITF/1/

File: config/jobs/kubernetes/release/release-config.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ presubmits:
       testgrid-alert-email: release-managers+alerts@kubernetes.io
       testgrid-num-columns-recent: '30'
   - name: pull-release-image-kube-cross
-    optional: true
+    optional: false
```
Comment (justaugustus) [inline config/jobs/kubernetes/release/release-config.yaml:92]:
```suggestion
```

You can drop all of these lines, since they default to `false`.

**Review by justaugustus — CHANGES_REQUESTED** [review]
@cpanato -- Just a few nits

File: config/jobs/kubernetes/release/release-config.yaml (hunk 2 above)
Comment (cpanato) [reply config/jobs/kubernetes/release/release-config.yaml:92]: 👍

File: config/jobs/image-pushing/releng/k8s-staging-releng.yaml (hunk 1 above)
Comment (cpanato) [reply config/jobs/image-pushing/releng/k8s-staging-releng.yaml]: ahh! make sense

done

**Pull request merged**